src/_nebari/provider/cloud/amazon_web_services.py

```diff
@@ -121,6 +121,27 @@ def instances(region: str) -> Dict[str, str]:
     return {t: t for t in instance_types}
 
 
+@functools.lru_cache()
```

Comment:
This is stylistic, so it is up to you whether to adopt this. I would probably add a `Kms_Key_Info` class or something.

```python
@dataclass
class Kms_Key_Info:
    Arn: str
    KeyUsage: str
    KeySpec: str
    KeyManager: str
```

The advantage here is more clarity in your type annotations. Rather than returning `Dict[str, dict] ` you can return `Dict[str, Kms_Key_Info]` which is clearer what is actually returned.

Reply:
@dcmcand 
Updated, per your suggestion